crates/fuel-core/src/graphql_api/required_fuel_block_height_extension.rs

```diff
@@ -98,33 +109,51 @@ impl Extension for RequiredFuelBlockHeightInner {
         let current_block_height = view.latest_block_height();
 
         if let Some(required_block_height) = self.required_height.get() {
-            if let Ok(current_block_height) = current_block_height {
-                if *required_block_height > current_block_height {
+            if let Ok(mut current_block_height) = current_block_height {
+                // Check first that the node is not out of sync
+                let minimum_block_height: BlockHeight = required_block_height
+                    .saturating_sub(self.tolerance_threshold)
+                    .into();
+                if current_block_height < minimum_block_height {
                     let (line, column) = (line!(), column!());
-                    let mut response = Response::from_errors(vec![ServerError::new(
-                        format!(
-                            "The required fuel block height is higher than the current block height. \
-                            Required: {}, Current: {}",
-                            // required_block_height: &BlockHeight, dereference twice to get the 
-                            // corresponding value as u32. This is necessary because the Display 
-                            // implementation for BlockHeight displays values in hexadecimal format.
-                            **required_block_height,
-                            // current_fuel_block_height: BlockHeight, dereference once to get the 
-                            // corresponding value as u32.
-                            *current_block_height
-                        ),
-                        Some(Pos {
-                            line: line as usize,
-                            column: column as usize,
-                        }),
-                    )]);
-
-                    response.extensions.insert(
-                        CURRENT_FUEL_BLOCK_HEIGHT.to_string(),
-                        Value::Number((*current_block_height).into()),
+                    return error_response(
+                        required_block_height,
+                        &current_block_height,
+                        line,
+                        column,
                     );
+                }
 
-                    return response
+                // If the node is lagging behind w.r.t. the required fuel block height,
+                // but within an acceptable interval, wait until the current block height
+                // matches the required fuel block height.
+                while current_block_height < *required_block_height {
+                    tokio::time::sleep(SLEEP_DURATION).await;
+                    match view.latest_block_height() {
```

Comment:
It would be nice if we could subscribe for the height that we need and notify all dependent tasks.

Reply:
BTW, without https://github.com/FuelLabs/fuel-core/pull/2646 it is very expensive operation, and do it for each query can be a lot. Before releasing RPC consistancy, we need merge this optimization